spec/javascript/unit/basetype_spec.js

```diff
@@ -67,3 +67,30 @@ describe('basetype Interval', () => {
     expect(() => { (new Interval()).cast({ low: { value: '30', unit: 'mcg' }, high: { value: '60', unit: 'mg' } }); }).toThrow();
   });
 });
+
+describe('basetype Code', () => {
+  it('can create a Code from valid hash', () => {
+    const code = (new Code()).cast({code: '12345', system: '1.2.3.4.5.6'});
+    expect(code.isCode).toBe(true);
+  });
+
+  it('can create a Code from valid hash with version and display', () => {
+    const code = (new Code()).cast({code: '12345', system: '1.2.3.4.5.6', version: '1', display: 'test'});
+    expect(code.isCode).toBe(true);
+  });
+
+  it('throws if invalid Code passed to cast', () => {
+    expect(() => { (new Code()).cast({bad_code_key: '12345', system: '1.2.3.4.5.6'}); }).toThrow();
+  });
+
+  it('casts null code to null', () => {
+    const code = (new Code()).cast(null);
+    expect(code).toBeNull()
+  });
+
+  it('cast returns code that is already a code', () => {
+    const code = (new Code()).cast({code: '12345', system: '1.2.3.4.5.6'});
+    const castedCode = (new Code()).cast(code);
+    expect(castedCode.isCode).toBe(true);
```

Comment:
Make sure it is the same one.
```suggestion
    expect(castedCode).toBe(code);
```